### ISSUE TITLE
Create previews for any merges going into `main`.

### ISSUE THREAD

**Issue by johnnymatthews** [entropyxyz/entropy-docs#82]
This issue proposes implementing a system to generate preview environments or deployments for any pull request attempting to merge into the `main` branch.

- Previews allow for manual testing and visual inspection before merging into `main`, reducing the risk of regressions or unexpected behavior.
- Devs can identify and address issues early in the development cycle.
- Preview environments isolate changes, preventing potential issues from impacting the production branch.

## Tasks

- [x] Figure out if this is possible with GitHub pages.
    - [x] Yes: Great!
    - [ ] No: Figure out some system with something like Cloudflare Pages.
- [ ] Implement chosen solution.

**Comment by johnnymatthews** [comment]
Currently waiting on https://github.com/entropyxyz/entropy-docs/issues/153
